pyro/optim/lr_scheduler.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+
+import pyro
+from pyro.params import module_from_param_with_module_name, user_param_name
+
+
+class PyroLRScheduler(object):
```

Comment:
given the design you've chosen why is this a separate object instead of subclass of `PyroOptim` or the like?

Reply:
solely to follow the namespace convention of [pytorch's scheduler](https://pytorch.org/docs/stable/optim.html?#torch.optim.lr_scheduler.LambdaLR) (note the LR classes extend `_LRScheduler` not `Optimizer` ). i could make it a `PyroOptim` but then that blurs the distinction between schedulers and optimizers in our interface (which may be fine? not sure)

Reply:
@neerajprad that would require changing the existing interface, as currently our Pyro Optims automatically wrap the torch class, so users use `pyro.optim.Adam` instead of `pyro.optim.PyroOptim(torch.optim.Adam, ...)` which is what i believe you're suggesting, which is also required by the scheduler. what martin is suggesting is to just make the scheduler `pyro.optim.LambdaLR` instead of `pyro.optim.lr_scheduler.LambdaLR`. i think we can do what youre suggesting but that would require a backwards change to the existing optimizers

Reply:
One question is whether we want the scheduler to be automatically called in `SVI.step`.  The PyTorch idiom is to call them per epoch rather than per update, suggesting we should perhaps keep it separate from `PyroOptim` as in the current design.